### PR TITLE
Add Aspeed Ast10x0 reset

### PR DESCRIPTION
**Pull request by aspeeddylan**
Add support of reset driver for Aspeed AST10x0 series SOC

### CODE REVIEW

**Review by carlescufi — DISMISSED** [review]
This LGTM after a quick passs

File: drivers/reset/reset_ast10x0.c
Comment (gmarull) [inline drivers/reset/reset_ast10x0.c:33]: please remove usage of such macros, this was deleted from all drivers a long time ago.

File: drivers/reset/reset_ast10x0.c
Comment (gmarull) [inline drivers/reset/reset_ast10x0.c:60]: return from here, this ignores error handling. If your platform can't error, do `(void)call()`. Same for others.

File: drivers/reset/reset_ast10x0.c
Comment (gmarull) [inline drivers/reset/reset_ast10x0.c:78]: ret is not a boolean, use `< 0` or `!= 0`

File: drivers/reset/reset_ast10x0.c
Comment (gmarull) [inline drivers/reset/reset_ast10x0.c:94]: what about line_toggle? any reason it is not supported?

**Review by gmarull — CHANGES_REQUESTED** [review]
overall lgtm, please check my comments

File: drivers/reset/reset_ast10x0.c
Comment (aspeeddylan) [reply drivers/reset/reset_ast10x0.c]: Thank you for your comment.  I removed the macros in the latest PR.

File: drivers/reset/reset_ast10x0.c
Comment (aspeeddylan) [reply drivers/reset/reset_ast10x0.c]: Revised.  The return value is from syscon_write_reg now.

File: drivers/reset/reset_ast10x0.c
Comment (aspeeddylan) [reply drivers/reset/reset_ast10x0.c]: Fixed in the latest PR.  It should be `ret == 0` since `status` is available only if the read of the syscon register succeed.

File: drivers/reset/reset_ast10x0.c
Comment (aspeeddylan) [reply drivers/reset/reset_ast10x0.c:103]: `line_toggle` is added in the latest PR.  Thank you for your review.

**Review by gmarull — APPROVED** [review]

**Comment by aspeeddylan** [comment]
> This LGTM after a quick passs

@carlescufi your approval was dismissed by the force push. Could your review it again? Thank you.

**Comment by aspeeddylan** [comment]
Is there anyone who can complete a second review of this?

**Review by carlescufi — APPROVED** [review]
LGTM

**Pull request merged**
